[IMP] account_multicompany_code: fixes pending from migration T#54296

LuisMzz opened the pull request:
- Split the models in individual files
- Add the code field to tree and kanban view
- Define a setUpClass method for tests
- Set a SQL constraint "code_unique" for efficiency
- Rename view: company_view -> res_company_views.xml

ghost commented:
@luisg123v Please review

luisg123v commented on account_multicompany_code/models/__init__.py:
```diff
@@ -1,2 +1,3 @@
 from . import res_company
 from . import account
+from . import journal
```
```suggestion
from . import account_journal
```

And rename account to its corresponding model.

Nitpick: also sort alphabetically

luisg123v commented on account_multicompany_code/models/res_company.py:
```diff
@@ -1,14 +1,8 @@
 
 class Company(models.Model):
     _inherit = "res.company"
+    _sql_constraints = [("unique_code", "UNIQUE(code)", "Code must be unique")]
```
- Move below fields, as stated in
[the guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#id40)
- Add a trailing comma after the constraint, so it's formatted one constraint per line, i.e.:
  ```
  _sql_constraints = [
      (constraint1),
  ]
  ```

luisg123v commented on account_multicompany_code/views/res_company_views.xml:
```diff
@@ -0,0 +1,40 @@
+    </record>
+
+    <record id="view_company_tree" model="ir.ui.view">
+        <field name="name">qf_common.company.tree</field>
```
Fix name, I'd recommend:

`res.company.tree.inherit.account.multicompany`

I just noticed the existing view was also wrong, so please fix also that one

luisg123v reviewed (changes requested):
Please squash, 2nd commit doesn't need to be on a separate commit IMHO, as it's part of the fixes.

luisg123v reviewed (changes requested):
- Fix name in commit autor, it should be your actual name, not your user name
- Remove username from PR's title
- Title: corrections -> fixes. It could be fixes pending from migration

Commit:
[IMP] account_multicompany_code: fixes pending from migration

- Split the models in individual files
- Add the code field to tree and kanban view
- Define a setUpClass method for tests
- Set a SQL constraint "code_unique" for efficiency
- Rename file: company_view -> res_company_views.xml

luisg123v commented on account_multicompany_code/models/res_company.py:
```diff
@@ -1,14 +1,11 @@
-            if record.code and record.search([("code", "=", record.code), ("id", "!=", record.id)]):
-                raise ValidationError(_("Code must be unique"))
+    _sql_constraints = [
+        ("unique_code", "UNIQUE(code)", "Code must be unique"),
```
I forgot to mention, please update translations, because the old message doesn't exist anymore and this is a new one.

LuisMzz replied:
I already update the translations

ghost commented:
@luisg123v Please review
